File: LeetCode/MockInterviews/SpiralMatrix.cs
namespace LeetCode
{
    class SpiralMatrix
    {
        static void PrintMatrix(int[,] matrix, int size)
        {
            for (int i = 0; i < size; i++)
            {
                System.Console.Write("[");
                for (int j = 0; j < size; j++)
                {
                    System.Console.Write(" "+matrix[i, j]+" ");
                }
                System.Console.WriteLine("]");
            }
        }

        static int[,] GenerateMatrix(int n)
        {
            int[,] matrix = new int[n,n];
            int sideSize = n - 1;
            int number = 1;
            int i = 0;
            int j = 0;
            int loop = 0;
            // fill first row
            for ( ; j < n; j++)
            {
                matrix[i, j] = number++;
            }
            j--;
            // loop for sideSize
            for ( ; sideSize > 0; sideSize--)
            {
                for ( loop = 0 ; loop < sideSize; loop++)
                {
                    matrix[++i, j] = number++;
                }
                for ( loop = 0; loop < sideSize; loop++)
                {
                    matrix[i, --j] = number++;
                }
                sideSize--;
                for ( loop = 0; loop < sideSize; loop++)
                {
                    matrix[--i, j] = number++;
                }
                for ( loop = 0; loop < sideSize; loop++)
                {
                    matrix[i, ++j] = number++;
                }
            }
            return matrix;
        }

        static void Main(string[] args)
        {
            int size = 6;
            int[, ] result = GenerateMatrix(size);
            PrintMatrix(result, size);
        }
    }
}

File: LeetCode/MockInterviews/Python/spiral_matrix.py
from pprint import pprint

def print_vars(**kwargs):
    pprint(kwargs)

def spiral_matrix(size):
    matrix = [[0 for _ in range(size)] for _ in range(size)]

    # first full row
    num = 1
    for x in range(size):
        matrix[0][x] = num
        num += 1
    corner = size-1
    i = 0
    j = size-1
    while corner > 0:
        for _ in range(corner):
            i += 1
            matrix[i][j] = num
            num += 1
        for _ in range(corner):
            j -= 1
            matrix[i][j] = num
            num += 1
        corner -= 1
        for _ in range(corner):
            i -= 1
            matrix[i][j] = num
            num += 1
        for _ in range(corner):
            j += 1
            matrix[i][j] = num
            num += 1
        corner -= 1
    pprint(matrix, indent=4)


if __name__ == '__main__':
    spiral_matrix(4)
